### PR TITLE
Nrcan nv

### PR DESCRIPTION
**Pull request by srgilani**

### CODE REVIEW

File: lib/openstudio-standards/standards/necb/BTAPPRE1980/btap_pre1980.rb [hunk 1]
```diff
@@ -86,7 +86,13 @@ def model_apply_standard(model:,
                            pv_ground_total_area_pv_panels_m2: nil ,
                            pv_ground_tilt_angle: nil,
                            pv_ground_azimuth_angle: nil,
-                           pv_ground_module_description: nil
+                           pv_ground_module_description: nil,
+                           nv_type: nil,
```
Comment (phylroy) [inline lib/openstudio-standards/standards/necb/BTAPPRE1980/btap_pre1980.rb:90]: Where would a designer get these inputs? Are there defaults?

File: lib/openstudio-standards/standards/necb/ECMS/nv.rb [hunk 2]
```diff
@@ -0,0 +1,187 @@
+class ECMS
+
+  def apply_nv(model:, nv_type:, nv_comfort_model:, nv_opening_fraction:, nv_Tout_min:, nv_Delta_Tin_Tout:)
+
+    ##### If any of users' inputs are nil/false, do nothing.
```
Comment (phylroy) [inline lib/openstudio-standards/standards/necb/ECMS/nv.rb:5]: Do you have an option for 'NECB_Default'

File: lib/openstudio-standards/standards/necb/ECMS/nv.rb [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+            if thermal_zone.thermostat.get.to_ThermostatSetpointDualSetpoint.get.heatingSetpointTemperatureSchedule.is_initialized ||
+                thermal_zone.thermostat.get.to_ThermostatSetpointDualSetpoint.get.coolingSetpointTemperatureSchedule.is_initialized
+              zone_thermostat = thermal_zone.thermostatSetpointDualSetpoint.get
+              zone_clg_thermostat_sch = zone_thermostat.coolingSetpointTemperatureSchedule.get
+              zone_htg_thermostat_sch = zone_thermostat.heatingSetpointTemperatureSchedule.get
+
+              ##### Create schedule for max Tin to have NV on the basis of cooling setpoint temperature for default day
+              zone_clg_thermostat_sch_name = zone_clg_thermostat_sch.name
+              zone_clg_sp_schedule = zone_clg_thermostat_sch.to_ScheduleRuleset.get
+              puts zone_clg_sp_schedule.name
```
Comment (phylroy) [inline lib/openstudio-standards/standards/necb/ECMS/nv.rb:45]: What is this puts here for?

File: lib/openstudio-standards/standards/necb/ECMS/nv.rb [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+              i = 0.0
+              max_Tin_schedule_defaultDay_times.each do |time|
+                max_Tin_schedule_defaultDay.addValue(time, max_Tin_schedule_defaultDay_values_adjusted[i])
+                i += 1.0
+              end
+
+              ##### Create schedule for min Tin to have NV on the basis of cooling setpoint temperature for default day
+              zone_htg_thermostat_sch_name = zone_htg_thermostat_sch.name
+              zone_htg_sp_schedule = zone_htg_thermostat_sch.to_ScheduleRuleset.get
+              puts zone_htg_sp_schedule.name
```
Comment (phylroy) [inline lib/openstudio-standards/standards/necb/ECMS/nv.rb:64]: same...remove the puts unless it tell the user something.

File: lib/openstudio-standards/standards/necb/ECMS/nv.rb [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+        end
+        # puts "#{space.name.to_s} has #{number_of_windows} window(s)"
+        oa_per_person_normalized_by_number_of_windows = outdoor_air_flow_per_person/number_of_windows
+        oa_per_floor_area_normalized_by_number_of_windows = outdoor_air_flow_per_floor_area/number_of_windows
+
+        ##### Add NV in each space that has window(s) using two objects: "ZoneVentilation:DesignFlowRate" and "ZoneVentilation:WindandStackOpenArea"
+        space.surfaces.sort.each do |surface|
+          surface.subSurfaces.sort.each do |subsurface|
+            # puts subsurface.name.to_s
+            if (subsurface.subSurfaceType == 'OperableWindow' || subsurface.subSurfaceType == 'FixedWindow') && subsurface.outsideBoundaryCondition == 'Outdoors'  #TODO: Question: Should I change window type to only 'OperableWindow'? For testing purposes, I have included the window type of 'FixedWindow' as well.
```
Comment (phylroy) [inline lib/openstudio-standards/standards/necb/ECMS/nv.rb:103]: To answer your question. Add to both as you did. You can remove the comment.

File: lib/openstudio-standards/standards/necb/ECMS/nv.rb [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+                # (Ref: E+ I/O) "Effective Angle is the angle in degrees counting from the North clockwise to the opening outward normal."
+                zn_vent_wind_and_stack.setEffectiveAngle(window_azimuth_deg)
+                zn_vent_wind_and_stack.setMinimumIndoorTemperatureSchedule(min_Tin_schedule)
+                zn_vent_wind_and_stack.setMaximumIndoorTemperatureSchedule(max_Tin_schedule)
+                zn_vent_wind_and_stack.setMinimumOutdoorTemperature(nv_Tout_min)
+                zn_vent_wind_and_stack.setMaximumOutdoorTemperatureSchedule(max_Tin_schedule)
+                zn_vent_wind_and_stack.setDeltaTemperature(nv_Delta_Tin_Tout)
+                zone_hvac_equipment_list.addEquipment(zn_vent_wind_and_stack)
+
+              # elsif nv_comfort_model == 'Adaptive_Model'  #TODO: QUESTION: do we want to include adaptive thermal comfort model (ASHRAE 55)?
```
Comment (phylroy) [inline lib/openstudio-standards/standards/necb/ECMS/nv.rb:156]: Not sure.. Does this control when users open the window? Would it interfere with the deltaT control?

**Review by phylroy — CHANGES_REQUESTED** [review]

File: lib/openstudio-standards/standards/necb/BTAPPRE1980/btap_pre1980.rb (hunk 1 above)
Comment (srgilani) [reply lib/openstudio-standards/standards/necb/BTAPPRE1980/btap_pre1980.rb]: Added. See my revisions here:
lib/openstudio-standards/standards/necb/ECMS/nv.rb
lib/openstudio-standards/standards/necb/BTAPPRE1980/btap_pre1980.rb

File: lib/openstudio-standards/standards/necb/ECMS/nv.rb (hunk 2 above)
Comment (srgilani) [reply lib/openstudio-standards/standards/necb/ECMS/nv.rb]: Added.

File: lib/openstudio-standards/standards/necb/ECMS/nv.rb (hunk 3 above)
Comment (srgilani) [reply lib/openstudio-standards/standards/necb/ECMS/nv.rb]: Removed.

File: lib/openstudio-standards/standards/necb/ECMS/nv.rb (hunk 4 above)
Comment (srgilani) [reply lib/openstudio-standards/standards/necb/ECMS/nv.rb]: Removed.

File: lib/openstudio-standards/standards/necb/ECMS/nv.rb (hunk 5 above)
Comment (srgilani) [reply lib/openstudio-standards/standards/necb/ECMS/nv.rb]: Removed the comment.

File: lib/openstudio-standards/standards/necb/ECMS/nv.rb (hunk 6 above)
Comment (srgilani) [reply lib/openstudio-standards/standards/necb/ECMS/nv.rb]: Removed the nv_comfort_model argument and the relevant if-clause.

**Pull request closed without merging**